lang/ghc/Portfile

```diff
@@ -61,179 +52,195 @@ platform powerpc {
-                set sLib /usr/lib/libSystem.B.dylib
-                set lLib ${prefix}/lib/libMacportsLegacySystem.B.dylib
-                system "install_name_tool -change ${sLib} ${lLib} ${f} > /dev/null 2>&1 ; true"
+    # If legacy support is active, patch the bootstrap binaries to redirect libSystem.B.dylib
+    # to MacPorts wrapped implementation which reexports the system library, adding
+    # each missing in each OS.
+    if {${os.platform} eq "darwin" && ${os.major} <= [legacysupport::get_newest_darwin_with_missing_symbols]} {
+        depends_lib-append [legacysupport::get_dependency]
+        post-extract {
+            foreach f [ exec find ${workpath}/ -type f -and -name "*.dylib" -or -name "*ghc*" -and -not -name "*stderr"] {
+                if {[string match "application/*" [lindex [exec /usr/bin/file --mime-type ${f}] end]]} {
+                    set sLib /usr/lib/libSystem.B.dylib
+                    set lLib ${prefix}/lib/libMacportsLegacySystem.B.dylib
+                    system "install_name_tool -change ${sLib} ${lLib} ${f} > /dev/null 2>&1 ; true"
+                }
             }
         }
     }
-}
 
-# Build requires C11
-compiler.c_standard 2011
-# On OSX 10.{8,9} with clang 503, 600
-#   :info:build dtrace: failed to open td=gnu11/usr/bin/clang -std=gnu11 -E ...
-compiler.blacklist-append {clang < 700}
-
-# use these to specify python versions, python3 required
-# use ${prefix}/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/python-1.0.tcl
-set python3_version 39
-set python3_branch [string index ${python3_version} 0].[string range ${python3_version} 1 end]
-
-set srcpath ${workpath}/src
-
-# get the objdump rpath names that match ${workpath} in the binary
-proc get_workpath_list { binname workpath_destdir_pattern } {
-    return [exec bash -c "otool -l \"${binname}\" 2>/dev/null \
-        | grep -E -A 2 -e '\[\[:space:]]+cmd\[\[:space:]]+LC_RPATH' \
-        | grep -E -e '\[\[:space:]]+path\[\[:space:]]+' \
-        | grep -E -o -e '(${workpath_destdir_pattern}\[^ ]+)' || true"]
-}
-
-# fix ${workpath} rpath dirs
-proc fix_workpath_rpath { binname workpath_destdir_pattern libdir_glob rpath_re } {
-    global destroot
-    set workpath_list \
-        [get_workpath_list ${binname} ${workpath_destdir_pattern}]
-    foreach path [lreverse ${workpath_list}] {
-        system "install_name_tool -delete_rpath \"${path}\" \"${binname}\" || true"
+    # Build requires C11
+    compiler.c_standard 2011
+    # On OSX 10.{8,9} with clang 503, 600
+    #   :info:build dtrace: failed to open td=gnu11/usr/bin/clang -std=gnu11 -E ...
+    compiler.blacklist-append {clang < 700}
+
+    # use these to specify python versions, python3 required
+    # use ${prefix}/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/python-1.0.tcl
+    set python3_version 39
```

Comment:
Python 3.10 is the current MacPorts' default, can that be used?

Reply:
Yes, I'll add that.